draft-ietf-tls-esni.md

```diff
@@ -784,12 +784,18 @@ If the server rejects ECH, the client proceeds with the handshake,
 authenticating for ECHConfig.contents.public_name as described in
 {{auth-public-name}}. If authentication or the handshake fails, the client MUST
 return a failure to the calling application. It MUST NOT use the retry
-configurations.
+configurations. It MUST NOT treat this as a secure signal to
+disable ECH.
 
 Otherwise, if both authentication and the handshake complete successfully, the
-client MUST abort the connection with an "ech_required" alert. It then
-processes the "retry_configs" field from the server's "encrypted_client_hello"
-extension.
+client then processes the server's response as follows.
```

Comment:
I understand the intent here, but this seems somewhat out of order. The checks below happen _before_ the handshake completes, but this text suggests they happen afterwards. Perhaps we can rewrite this as follows?

```
Otherwise, in authenticating the server and completing the handshake, 
the client then processes the server's response as follows.
```

Reply:
Good catch - this ordering issue is one I was trying to expunge during the first pass, and it has now crept back in with the decode_error processing.  There's possibly another instance of this in the suggestion for the comment below, depending on how the question I'm asking there resolves.